lib/galaxy/visualization/plugins/resource_parser.py

```diff
@@ -197,19 +197,19 @@ def parse_parameter(self, trans, expected_param_data, query_param,
         # db models
         elif param_type == 'visualization':
             # ?: is this even used anymore/anywhere?
-            decoded_visualization_id = self._decode_id(query_param)
+            decoded_visualization_id = trans.security.decode_id(query_param, object_name=param_type)
             parsed_param = self.managers.visualization.get_accessible(decoded_visualization_id, trans.user)
 
         elif param_type == 'dataset':
-            decoded_dataset_id = self._decode_id(query_param)
+            decoded_dataset_id = trans.security.decode_id(query_param, object_name=param_type)
             parsed_param = self.managers.hda.get_accessible(decoded_dataset_id, trans.user)
 
         elif param_type == 'hda_or_ldda':
             encoded_dataset_id = query_param
             # needs info from another param...
             hda_ldda = param_modifiers.get('hda_ldda')
             if hda_ldda == 'hda':
-                decoded_dataset_id = self._decode_id(encoded_dataset_id)
+                decoded_dataset_id = trans.security.decode_id(query_param, object_name="dataset")
                 parsed_param = self.managers.hda.get_accessible(decoded_dataset_id, trans.user)
             else:
                 parsed_param = self.managers.ldda.get(trans, encoded_dataset_id)
```

Comment:
Now that I look at this code, it seems there is no `ldda` defined in the managers [here](https://github.com/galaxyproject/galaxy/pull/11856/files#diff-34ed9d4f6edf6d83e2749c0c4ebb1234af2f40a9c18b6c1013115f32aa8508b3L46)?

Reply:
Indeed.